malaria24/ona/models.py

```diff
@@ -666,16 +671,27 @@ def get_data(self):
                 #       old format.
                 birth_date = datetime.strptime(self.date_of_birth, '%y%m%d')
 
+            valid_prefix = '[+27]'
+            if re.match('^' + valid_prefix + '*([0-9]{9})$', self.reported_by):
```

Comment:
Looks like your duplicating this part for reported_by and msisdn, I would rather move it into a function and just call the function.

Have a look at this: `https://github.com/praekelt/hellomama-registration/blob/develop/hellomama_registration/utils.py#L81`

Reply:
will do, thank you